CHANGELOG.md

```diff
@@ -100,6 +100,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed issue with pickling `CSVLogger` after a call to `CSVLogger.save` ([#10388](https://github.com/PyTorchLightning/pytorch-lightning/pull/10388))
 
+
+- Disabled step logging on epoch level hooks ([#10409](https://github.com/PyTorchLightning/pytorch-lightning/pull/10409))
```

Comment:
should be moved to the "changed" section or mention what was fixed, right? 

It's not clear what the current behavior was. Can you describe it in the PR description or link an issue?

Reply:
updated the description.
> should be moved to the "changed" section or mention what was fixed, right?

I was wondering this too. But since this is something unexpected and can happen from user's side I thought making it as a bugfix might make more sense because we don't want this and this was something missed from our side.

Reply:
So I assume you tested it out and there were unintended side effects? Then I think the  changelog should read something like


Fixed:
- Fixed an issue with logging with `on_step=True` in epoch-level hooks causing unintended side-effects. Logging with `on_step=True` in epoch-level hooks will now correctly raise an error.


:)

Reply:
sure.. will update.
btw found this while working on something else.. It can cause side effects but haven't tried yet.